README.md

```diff
@@ -16,6 +15,38 @@ You can install the API Library with the following command:
 composer require mautic/api-library
 ```
 
+N.B. Make sure you have installed a PSR-18 HTTP Client before you install this package or install one at the same time e.g. `composer require mautic/api-library guzzlehttp/guzzle:^7.3`.
+
+### HTTP Client
+
+We are decoupled from any HTTP messaging client with the help of [PSR-18 HTTP Client](https://www.php-fig.org/psr/psr-18/). This requires an extra package providing [psr/http-client-implementation](https://packagist.org/providers/psr/http-client-implementation). To use Guzzle 7, for example, simply require `guzzlehttp/guzzle`:
+
+``` bash
+composer require guzzlehttp/guzzle:^7.3
+```
+
+The installed HTTP Client is auto-discovered using [php-http/discovery](https://packagist.org/providers/php-http/discovery), but you can also provide your own HTTP Client if you like.
+
+```php
+<?php
+
+// Bootup the Composer autoloader
+include __DIR__ . '/vendor/autoload.php';  
+
+use GuzzleHttp\Client;
+use Mautic\Auth\ApiAuth;
+
+// Initiate an HTTP Client
+$httpClient = new Client([
+    'timeout'  => 10,
+]);
+
+// Initiate the auth object
+$initAuth = new ApiAuth();
+$auth     = $initAuth->newAuth($settings)->withHttpClient($httpClient);
```

Comment:
I'm thinking if this wouldn't be more dependency injection way:
```suggestion
$initAuth = new ApiAuth($httpClient);
$auth     = $initAuth->newAuth($settings);
```

Reply:
Yes, that'd be a better solution, thanks!